lib/default/TasmotaSerial-3.6.0/src/TasmotaSerial.cpp

```diff
@@ -152,7 +152,7 @@ void TasmotaSerial::setTransmitEnablePin(int tx_enable_pin) {
 bool TasmotaSerial::freeUart(void) {
   for (uint32_t i = SOC_UART_NUM -1; i >= 0; i--) {
     if (0 == bitRead(tasmota_serial_uart_bitmap, i)) {
-      m_uart = i;
+      m_uart = uart_port_t(i);
```

Comment:
Oh my. Not sure if this should be changed like shown. Will need investigation as the m_uart is just supposed to be a bitmap for Tasmota internally.... I'll dive in.

Reply:
No, in the code we set  the bitmap `tasmota_serial_uart_bitmap` with the value of `m_uart`. So the latter is not a bitmap and will be used in other parts directly to call `uart_set_rx_full_threshold(m_uart, <value>);`. 

`uart_port_t `was a typedef'd integer before and will become an enum with IDF 5.2.
`uart_set_rx_full_threshold` only will accept `uart_port_t` type after 5.2.

Reply:
Thx. Correct